addons/binding/org.openhab.binding.fritzaha/src/main/java/org/openhab/binding/fritzaha/handler/FritzAhaBridgeHandler.java

```diff
@@ -0,0 +1,286 @@
+				+ device.toString());
+		if (device.isTempSensor()) {
+			Channel channel = thing.getChannel(CHANNEL_TEMP);
+			this.updateState(channel.getUID(), new DecimalType(device
+					.getTemperature().getCelsius()));
+		}
+		if (device.isPowermeter()) {
+			Channel channelEnergy = thing.getChannel(CHANNEL_ENERGY);
+			this.updateState(channelEnergy.getUID(), new DecimalType(device
+					.getPowermeter().getEnergy()));
+			Channel channelPower = thing.getChannel(CHANNEL_POWER);
+			this.updateState(channelPower.getUID(), new DecimalType(device
+					.getPowermeter().getPower()));
+		}
+		if (device.isSwitchableOutlet()) {
+			Channel channel = thing.getChannel(CHANNEL_SWITCH);
+			if (device.getSwitch().getState().equals(SwitchModel.ON)) {
+				this.updateState(channel.getUID(), OnOffType.ON);
+			} else if (device.getSwitch().getState().equals(SwitchModel.OFF)) {
+				this.updateState(channel.getUID(), OnOffType.OFF);
+			} else {
+				logger.warn("unknown state " + device.getSwitch().getState()
+						+ " for channel " + channel.getUID());
+			}
+		}
+	}
+
+	/**
+	 * Builds a {@link ThingUID} from a device model. The UID is build from 
+	 * the {@link FritzAhaBindingConstants#BINDING_ID} and value of 
+	 * {@link DeviceModel#getProductName()} in which all characters NOT matching
+	 * the regex [^a-zA-Z0-9_] are replaced by "_".
+	 * 
+	 * @param device Discovered device model
+	 * @return ThingUID without illegal characters.
+	 */
+	public ThingUID getThingUID(DeviceModel device) {
+		ThingUID bridgeUID = this.getThing().getUID();
+		ThingTypeUID thingTypeUID = new ThingTypeUID(BINDING_ID, device
+				.getProductName().replaceAll("[^a-zA-Z0-9_]", "_"));
+
+		if (FritzAhaBindingConstants.SUPPORTED_DEVICE_THING_TYPES_UIDS.contains(thingTypeUID)) {
+			String thingName = device.getIdentifier()
+					.replaceAll("[^a-zA-Z0-9_]", "_");
+			ThingUID thingUID = new ThingUID(thingTypeUID, bridgeUID, thingName);
+			return thingUID;
+		} else {
+			return null;
+		}
+	}
+
+	/**
+	 * Start the polling.
+	 */
+	private synchronized void onUpdate() {
+		if (this.getThing() != null) {
+			if (pollingJob == null || pollingJob.isCancelled()) {
+				logger.debug("start polling job at intervall " + refreshInterval);
+				pollingJob = scheduler.scheduleWithFixedDelay(pollingRunnable, 1,
+						refreshInterval, TimeUnit.SECONDS);
+			} else {
+				logger.debug("pollingJob active");
+			}
+		} else {
+			logger.warn("bridge is null");
+		}
+	}
+	/**
+	 * Just logging - nothing to do.
+	 */
+	@Override
+	public void handleCommand(ChannelUID channelUID, Command command) {
+		logger.debug("update " + channelUID.getAsString() + " with "
+				+ command.toString());
+	}
+
+	/**
+	 * Called from {@link FritzahaWebInterface#authenticate()} to update 
+	 * the bridge status.
+	 * @param status Bridge status
+	 * @param statusDetail Bridge status detail
+	 * @param description Bridge status description
+	 */
+	public void setStatusInfo(ThingStatus status,
```

Comment:
I will remove the logging. The method is required because updateStatus is protected and not accessible from FritzahaWebInterface